tests: bsim: 10x the watchdog timeout

jori-nordic opened the pull request:
Why: CI runners are now paid less, so they work less fast :p

Seriously though, we are starting to get intermittent failures in CI because the tests run slightly slower than on the previous runners. This timeout value is set too conservatively, it should be okay to increase it.

Here's why:

We always start the bsim PHY with a given test run-time. The PHY will stop the simulation after that time.

This timeout value is another layer on top of this, and will abort all the spawned processes in case something goes very wrong.

jori-nordic committed:
tests: bsim: 10x the watchdog timeout

- Increase the timeout by a factor of 10.
- Remove the explicit value if the resulting timeout is < 300s which is
  the new default of the `Execute` function.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

Thalley commented on tests/bsim/bluetooth/host/adv/chain/tests_scripts/adv_chain.sh:
```diff
@@ -8,7 +8,7 @@ source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
 simulation_id="adv_chain"
 verbosity_level=2
-EXECUTE_TIMEOUT=10
+EXECUTE_TIMEOUT=100
```
Since this is a real-time watchdog that we rarely hit for the actual reasons we have it, can't we just default to 300 and remove all the script specific ones below 300?

jori-nordic replied:
done. I also applied it to the rest of the bsim tests

jori-nordic committed:
tests: bsim: 10x default watchdog timeout

We are starting to get intermittent failures in CI because the tests run
slightly slower than on the previous runners. This timeout value is set
too conservatively, it should be okay to increase it.

Here's why:

We always start the bsim PHY with a given test run-time. The PHY will stop
the simulation after that time.

This timeout value is another layer on top of this, and will abort all the
spawned processes in case something goes very wrong.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

alwa-nordic commented:
Should we add tag hot-fix?

alxelax commented on tests/bsim/bluetooth/mesh/_mesh_test.sh:
```diff
@@ -3,7 +3,7 @@
 
 source ${ZEPHYR_BASE}/tests/bsim/sh_common.source
 
-EXECUTE_TIMEOUT=600
```
Hasn't it been already fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/70087 2 weeks ago?
Does it require more time since that fix?

alxelax replied:
Yep, for the last 2 weeks it grew up in 20 times. It seems weird to me. Does CI become 20 times slower? PC from 90th?

jori-nordic replied:
pretty much :smile: 
![image](https://github.com/zephyrproject-rtos/zephyr/assets/68428270/38776d14-582c-4da1-8a72-6e26a1c124fb)
https://github.com/zephyrproject-rtos/infrastructure/issues/150

@aescolar correct me if I'm wrong, but it's my understanding it's the parallelization of the runs that is causing the timeouts on those older machines.

alxelax replied:
Ok, since this is watchdog. I close topic to not hair up it. :)

Thalley commented on tests/bsim/sh_common.source:
```diff
@@ -40,7 +40,7 @@ function check_program_exists() {
 
 function Execute() {
-  EXECUTE_TIMEOUT="${EXECUTE_TIMEOUT:-30}"
+  EXECUTE_TIMEOUT="${EXECUTE_TIMEOUT:-300}"
```
Didn't you mention that we very rarely hit this in CI, as an actual issue? 

When running local, the timeout is less of an issue, as a user (I assume) will typically abort it manually if it gets stuck within seconds. 

The problem that this PR fixes is that the majority of the timeouts we see are false positives. The current watchdog values are more of a hinder than a help.

jori-nordic replied:
what's to say we won't run multiple CI jobs on the same physical machine in the future? In that case they might be 10x slower.

@alwa-nordic you had something for timeout on actual CPU used time, what was it again? We should probably switch to that instead.

aescolar replied:
> Didn't you mention that we very rarely hit this in CI, as an actual issue?
> When running local, 

We hit it as a real problem some times, and some of us use it locally (I do) to run many tests in batch. And sometimes I'm met with stuck tests for one reason or another.

alwa-nordic replied:
> @alwa-nordic you had something for timeout on actual CPU used time, what was it again? We should probably switch to that instead.

`ulimit -t <cpu_time_in_seconds>` is a simple limit on CPU time. The limit is not shared between processes. Each new fork is given the full amount, AFAIK. This does not adjust for the speed of the CPU.

And it does not count IO-blocked time. This part is important, because the real-world-clock timeout scheme will protect against the bsim phy forever waiting for a device that never shows up. Counting CPU-time will not.

I think we should have a watchdog that kills the whole job if the simulation-time is not progressing.

aescolar replied:
> what's to say we won't run multiple CI jobs on the same physical machine in the future?

Running several jobs or VMs in a single physical machine is very normal, but oversubcribing CI runners beyond their number of cores would not be normal or wise. For this kind of jobs what I'd expect the admin would do is split a machine with say 32 cores, into 4 or 8 VMs with 8 or 4 cores each. In that case, more load in the other VMs will slow down a bit your VM booked cores (because memory BW and TDP are shared), but your performance will not plummet to 1/10.

aescolar replied:
@jori-nordic I'm just asking that instead of increasing it to 10x you'd increase it to 2 or 3x

aescolar commented:
@jori-nordic 
Here you have a list of all tests, how long they take in my machine (with the same script as in CI, but running many more in parallel), and how long the EXECUTE_TIMEOUT is (assuming it is found trivially in the script, if it isn't the default is assumed and the next column has a 1). I'm guessing none of the ones which take <1s in my machine is causing trouble in CI with the default 30s timeout..

<!DOCTYPE html>

<html>

<body>

test | runtime | EXE_TIMEOUT | TIMEOUT found | ratio
-- | -- | -- | -- | --
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_slot_reservation.sh | 0.099 | 30 | 1 | 0.0033
tests/bsim/bluetooth/mesh/tests_scripts/comp_data/cdp1_encode_decode.sh | 0.318 | 30 | 1 | 0.0106
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_broadcast_unicast_seq.sh | 0.416 | 30 | 1 | 0.0138666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_broadcast_unicast.sh | 0.612 | 30 | 1 | 0.0204
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_broadcast_trans.sh | 0.734 | 30 | 1 | 0.0244666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_caps_no_rsp.sh | 0.881 | 30 | 1 | 0.0293666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_caps_all_rsp.sh | 0.941 | 30 | 1 | 0.0313666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_caps_partial_rsp.sh | 1.01 | 30 | 1 | 0.0336666666666667
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_ivu.sh | 0.691 | 30 | 1 | 0.0230333333333333
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_proxy_gatt.sh | 0.712 | 30 | 1 | 0.0237333333333333
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_interleave.sh | 0.457 | 30 | 1 | 0.0152333333333333
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_caps_cancelled.sh | 1.276 | 30 | 1 | 0.0425333333333333
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_cache.sh | 0.452 | 30 | 1 | 0.0150666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_broadcast_basic.sh | 1.441 | 30 | 1 | 0.0480333333333333
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_ivu_long_interval.sh | 0.59 | 30 | 1 | 0.0196666666666667
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_slot_idempotency.sh | 1.725 | 30 | 1 | 0.0575
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_proxy_node_id.sh | 0.999 | 30 | 1 | 0.0333
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_proxy_net_id_multi.sh | 0.516 | 30 | 1 | 0.0172
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_kr.sh | 0.684 | 30 | 1 | 0.0228
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_kr_long_interval.sh | 0.622 | 30 | 1 | 0.0207333333333333
tests/bsim/bluetooth/mesh/tests_scripts/friendship/establish.sh | 0.442 | 30 | 1 | 0.0147333333333333
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_dist_self_update.sh | 2.243 | 30 | 1 | 0.0747666666666667
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_invalid.sh | 0.888 | 30 | 1 | 0.0296
tests/bsim/bluetooth/mesh/tests_scripts/friendship/msg_va_collision.sh | 0.547 | 30 | 1 | 0.0182333333333333
tests/bsim/bluetooth/mesh/tests_scripts/friendship/lpn_disable.sh | 0.477 | 30 | 1 | 0.0159
tests/bsim/bluetooth/mesh/tests_scripts/friendship/msg_frnd.sh | 0.433 | 30 | 1 | 0.0144333333333333
tests/bsim/bluetooth/mesh/tests_scripts/friendship/establish_multi.sh | 0.782 | 30 | 1 | 0.0260666666666667
tests/bsim/bluetooth/mesh/tests_scripts/friendship/msg_group.sh | 1.074 | 30 | 1 | 0.0358
tests/bsim/bluetooth/mesh/tests_scripts/friendship/lpn_terminate_cb.sh | 0.498 | 30 | 1 | 0.0166
tests/bsim/bluetooth/mesh/tests_scripts/friendship/overflow.sh | 0.621 | 30 | 1 | 0.0207
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/proxy_adv_multi_subnet_coex.sh | 1.868 | 30 | 1 | 0.0622666666666667
tests/bsim/bluetooth/mesh/tests_scripts/friendship/msg_mesh.sh | 0.885 | 30 | 1 | 0.0295
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_dist_self_update_mult_targets.sh | 3.002 | 30 | 1 | 0.100066666666667
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/tx_cb_single.sh | 0.107 | 30 | 1 | 0.00356666666666667
tests/bsim/bluetooth/mesh/tests_scripts/friendship/lpn_loopback.sh | 0.717 | 30 | 1 | 0.0239
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/tx_cb_multi.sh | 0.373 | 30 | 1 | 0.0124333333333333
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/reverse_order.sh | 0.495 | 30 | 1 | 0.0165
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/random_order.sh | 0.124 | 30 | 1 | 0.00413333333333333
tests/bsim/bluetooth/mesh/tests_scripts/friendship/poll.sh | 1.12 | 30 | 1 | 0.0373333333333333
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/send_order.sh | 0.548 | 30 | 1 | 0.0182666666666667
tests/bsim/bluetooth/mesh/tests_scripts/friendship/re-establish.sh | 1.04 | 30 | 1 | 0.0346666666666667
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/disable.sh | 0.402 | 30 | 1 | 0.0134
tests/bsim/bluetooth/mesh/tests_scripts/friendship/msg_mesh_low_lat.sh | 1.376 | 30 | 1 | 0.0458666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_persistent_transfer_pull.sh | 3.822 | 30 | 1 | 0.1274
tests/bsim/bluetooth/mesh/tests_scripts/advertiser/proxy_mixin.sh | 0.91 | 30 | 1 | 0.0303333333333333
tests/bsim/bluetooth/mesh/tests_scripts/iv_index/iv_recovery.sh | 0.403 | 30 | 1 | 0.0134333333333333
tests/bsim/bluetooth/mesh/tests_scripts/iv_index/iv_deferring.sh | 0.515 | 30 | 1 | 0.0171666666666667
tests/bsim/bluetooth/mesh/tests_scripts/iv_index/iv_update.sh | 0.496 | 30 | 1 | 0.0165333333333333
tests/bsim/bluetooth/mesh/tests_scripts/scanner/wrong_packet_length.sh | 0.482 | 30 | 1 | 0.0160666666666667
tests/bsim/bluetooth/mesh/tests_scripts/scanner/invalid_ad_type.sh | 0.423 | 30 | 1 | 0.0141
tests/bsim/bluetooth/mesh/tests_scripts/replay_cache/rpl_frag.sh | 1.301 | 30 | 1 | 0.0433666666666667
tests/bsim/bluetooth/mesh/tests_scripts/replay_cache/replay_attack.sh | 1.806 | 30 | 1 | 0.0602
tests/bsim/bluetooth/mesh/tests_scripts/provision/ivu_flag_zero_duration.sh | 0.819 | 30 | 1 | 0.0273
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_remote_timeout.sh | 1.294 | 30 | 1 | 0.0431333333333333
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_all_targets_lost_on_caps_get.sh | 5.593 | 30 | 1 | 0.186433333333333
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_beacon_adv.sh | 4.713 | 30 | 1 | 0.1571
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_all_targets_lost_on_verify.sh | 5.87 | 30 | 1 | 0.195666666666667
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_all_targets_lost_on_metadata.sh | 5.954 | 30 | 1 | 0.198466666666667
tests/bsim/bluetooth/mesh/tests_scripts/priv_beacon/priv_proxy_net_id.sh | 4.774 | 30 | 1 | 0.159133333333333
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_adv_reprovision.sh | 2.505 | 30 | 1 | 0.0835
tests/bsim/bluetooth/mesh/tests_scripts/provision/ivu_flag_one_duration.sh | 1.401 | 30 | 1 | 0.0467
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_remote_client_server_same_dev.sh | 3.026 | 30 | 1 | 0.100866666666667
tests/bsim/bluetooth/mesh/tests_scripts/beacon/iv_update.sh | 1.213 | 30 | 1 | 0.0404333333333333
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_all_targets_lost_on_apply.sh | 7.439 | 30 | 1 | 0.247966666666667
tests/bsim/bluetooth/mesh/tests_scripts/heartbeat/sub_cb_api_all.sh | 1.956 | 30 | 1 | 0.0652
tests/bsim/bluetooth/mesh/tests_scripts/beacon/beacon_cache.sh | 0.808 | 30 | 1 | 0.0269333333333333
tests/bsim/bluetooth/mesh/tests_scripts/heartbeat/sub_cb_api_unicast.sh | 2.088 | 30 | 1 | 0.0696
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_adv_oob_auth_ignore_oob_pk.sh | 4.077 | 30 | 1 | 0.1359
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_remote_reprovision.sh | 4.587 | 30 | 1 | 0.1529
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_adv_no_oob.sh | 2.984 | 30 | 1 | 0.0994666666666667
tests/bsim/bluetooth/mesh/tests_scripts/beacon/invalid.sh | 1.847 | 30 | 1 | 0.0615666666666667
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_adv_oob_auth_ib_pk.sh | 4.582 | 30 | 1 | 0.152733333333333
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_adv_multi.sh | 4.502 | 30 | 1 | 0.150066666666667
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_all_targets_lost_on_update_get.sh | 8.722 | 30 | 1 | 0.290733333333333
tests/bsim/bluetooth/mesh/tests_scripts/beacon/key_refresh.sh | 1.457 | 30 | 1 | 0.0485666666666667
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_remote_parallel.sh | 5.285 | 30 | 1 | 0.176166666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_metadata_max_sdu.sh | 1.652 | 30 | 1 | 0.0550666666666667
tests/bsim/bluetooth/mesh/tests_scripts/beacon/kr_old_key.sh | 2.181 | 30 | 1 | 0.0727
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_adv_oob_auth_oob_pk.sh | 4.747 | 30 | 1 | 0.158233333333333
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp128_data_split_dfu.sh | 1.463 | 30 | 1 | 0.0487666666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp130_data_split.sh | 1.436 | 30 | 1 | 0.0478666666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp129_data_split_dfu.sh | 1.642 | 30 | 1 | 0.0547333333333333
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp2_data_split_dfu.sh | 1.275 | 30 | 1 | 0.0425
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp_data_max_sdu.sh | 1.23 | 30 | 1 | 0.041
tests/bsim/bluetooth/mesh/tests_scripts/op_agg/loopback.sh | 0.184 | 30 | 1 | 0.00613333333333333
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp2_data_split.sh | 1.331 | 30 | 1 | 0.0443666666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp0_data_split_dfu.sh | 1.365 | 30 | 1 | 0.0455
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp1_data_split.sh | 1.404 | 30 | 1 | 0.0468
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp130_data_split_dfu.sh | 1.362 | 30 | 1 | 0.0454
tests/bsim/bluetooth/mesh/tests_scripts/op_agg/full_status_msg_list.sh | 0.683 | 30 | 1 | 0.0227666666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_metadata_split.sh | 1.348 | 30 | 1 | 0.0449333333333333
tests/bsim/bluetooth/mesh/tests_scripts/op_agg/model_coex.sh | 0.85 | 30 | 1 | 0.0283333333333333
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp128_data_split.sh | 1.085 | 30 | 1 | 0.0361666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_no_rsp_block.sh | 10.487 | 30 | 1 | 0.349566666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp129_data_split.sh | 1.254 | 30 | 1 | 0.0418
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp0_data_split.sh | 1.295 | 30 | 1 | 0.0431666666666667
tests/bsim/bluetooth/mesh/tests_scripts/large_comp_data/get_comp1_data_split_dfu.sh | 1.132 | 30 | 1 | 0.0377333333333333
tests/bsim/bluetooth/mesh/tests_scripts/access/access_ext_sub_cap.sh | 0.099 | 30 | 1 | 0.0033
tests/bsim/bluetooth/mesh/tests_scripts/suspend/gatt_suspend_disable_resume.sh | 0.427 | 30 | 1 | 0.0142333333333333
tests/bsim/bluetooth/mesh/tests_scripts/sar/sar_cfg_persistent_storage.sh | 0.821 | 30 | 1 | 0.0273666666666667
tests/bsim/bluetooth/mesh/tests_scripts/persistence/provisioning.sh | 1.234 | 30 | 1 | 0.0411333333333333
tests/bsim/bluetooth/mesh/tests_scripts/beacon/multiple_netkeys.sh | 3.774 | 30 | 1 | 0.1258
tests/bsim/bluetooth/mesh/tests_scripts/sar/slow_transfer_test.sh | 1.088 | 30 | 1 | 0.0362666666666667
tests/bsim/bluetooth/mesh/tests_scripts/proxy_sol/sol_beacon_rcvd.sh | 0.777 | 30 | 1 | 0.0259
tests/bsim/bluetooth/mesh/tests_scripts/access/access_ext_sub.sh | 0.216 | 30 | 1 | 0.0072
tests/bsim/bluetooth/mesh/tests_scripts/suspend/gatt_suspend_resume.sh | 0.539 | 30 | 1 | 0.0179666666666667
tests/bsim/bluetooth/mesh/tests_scripts/access/access_cancel.sh | 0.379 | 30 | 1 | 0.0126333333333333
tests/bsim/bluetooth/mesh/tests_scripts/suspend/suspend_resume.sh | 0.724 | 30 | 1 | 0.0241333333333333
tests/bsim/bluetooth/mesh/tests_scripts/suspend/suspend_disable_resume.sh | 0.815 | 30 | 1 | 0.0271666666666667
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_slot.sh | 11.345 | 30 | 1 | 0.378166666666667
tests/bsim/bluetooth/mesh/tests_scripts/access/access_period_delayable.sh | 0.853 | 30 | 1 | 0.0284333333333333
tests/bsim/bluetooth/mesh/tests_scripts/transport/seg_fail.sh | 0.53 | 30 | 1 | 0.0176666666666667
tests/bsim/bluetooth/mesh/tests_scripts/access/access_transmit.sh | 0.928 | 30 | 1 | 0.0309333333333333
tests/bsim/bluetooth/mesh/tests_scripts/access/access_period.sh | 0.936 | 30 | 1 | 0.0312
tests/bsim/bluetooth/mesh/tests_scripts/persistence/cfg.sh | 2.194 | 30 | 1 | 0.0731333333333333
tests/bsim/bluetooth/mesh/tests_scripts/access/access_transmit_delayable.sh | 0.908 | 30 | 1 | 0.0302666666666667
tests/bsim/bluetooth/mesh/tests_scripts/sar/stress_test.sh | 1.802 | 30 | 1 | 0.0600666666666667
tests/bsim/bluetooth/mesh/tests_scripts/proxy_sol/sol_replay.sh | 1.764 | 30 | 1 | 0.0588
tests/bsim/bluetooth/mesh/tests_scripts/persistence/reprovisioning.sh | 2.331 | 30 | 1 | 0.0777
tests/bsim/bluetooth/mesh/tests_scripts/transport/seg_block.sh | 0.881 | 30 | 1 | 0.0293666666666667
tests/bsim/bluetooth/mesh/tests_scripts/transport/seg_concurrent.sh | 1.031 | 30 | 1 | 0.0343666666666667
tests/bsim/bluetooth/mesh/tests_scripts/transport/unicast_low_lat.sh | 0.847 | 30 | 1 | 0.0282333333333333
tests/bsim/bluetooth/mesh/tests_scripts/transport/loopback.sh | 0.998 | 30 | 1 | 0.0332666666666667
tests/bsim/bluetooth/mesh/tests_scripts/transport/unicast.sh | 1.058 | 30 | 1 | 0.0352666666666667
tests/bsim/bluetooth/mesh/tests_scripts/transport/loopback_group_low_lat.sh | 1.57 | 30 | 1 | 0.0523333333333333
tests/bsim/bluetooth/audio/test_scripts/ias.sh | 0.928 | 20 |   | 0.0464
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_remote_nppi_robustness.sh | 8.176 | 30 | 1 | 0.272533333333333
tests/bsim/bluetooth/mesh/tests_scripts/transport/unknown_app.sh | 1.35 | 30 | 1 | 0.045
tests/bsim/bluetooth/mesh/tests_scripts/persistence/access.sh | 3.18 | 30 | 1 | 0.106
tests/bsim/bluetooth/mesh/tests_scripts/transport/seg_ivu.sh | 1.654 | 30 | 1 | 0.0551333333333333
tests/bsim/bluetooth/mesh/tests_scripts/transport/fixed.sh | 1.625 | 30 | 1 | 0.0541666666666667
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_timeout.sh | 1.881 | 20 |   | 0.09405
tests/bsim/bluetooth/mesh/tests_scripts/transport/group.sh | 2.564 | 30 | 1 | 0.0854666666666667
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_mixed.sh | 13.943 | 30 | 1 | 0.464766666666667
tests/bsim/bluetooth/mesh/tests_scripts/transport/va.sh | 2.535 | 30 | 1 | 0.0845
tests/bsim/bluetooth/audio/test_scripts/vcp.sh | 1.534 | 20 |   | 0.0767
tests/bsim/bluetooth/mesh/tests_scripts/provision/pb_remote_pst_ncrp.sh | 10.803 | 30 | 1 | 0.3601
tests/bsim/bluetooth/mesh/tests_scripts/transport/loopback_group.sh | 2.94 | 30 | 1 | 0.098
tests/bsim/bluetooth/audio/test_scripts/micp.sh | 1.173 | 20 |   | 0.05865
tests/bsim/bluetooth/audio/test_scripts/bap_unicast_audio_acl_disconnect.sh | 1.748 | 20 |   | 0.0874
tests/bsim/bluetooth/audio/test_scripts/pacs_notify.sh | 0.865 | 200 |   | 0.004325
tests/bsim/bluetooth/audio/test_scripts/csip_encrypted_sirk.sh | 2.972 | 30 |   | 0.0990666666666667
tests/bsim/bluetooth/audio/test_scripts/cap_unicast.sh | 2.279 | 20 |   | 0.11395
tests/bsim/bluetooth/mesh/tests_scripts/transport/va_collision.sh | 4.251 | 30 | 1 | 0.1417
tests/bsim/bluetooth/audio/test_scripts/bap_bass_server_sync_client_rem.sh | 1.347 | 30 | 1 | 0.0449
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_inval.sh | 1.637 | 20 |   | 0.08185
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_friend.sh | 16.037 | 30 | 1 | 0.534566666666667
tests/bsim/bluetooth/audio/test_scripts/cap_broadcast.sh | 1.606 | 20 |   | 0.0803
tests/bsim/bluetooth/audio/test_scripts/gmap_broadcast_ac_14.sh | 2.875 | 60 |   | 0.0479166666666667
tests/bsim/bluetooth/audio/test_scripts/tmap.sh | 0.627 | 30 |   | 0.0209
tests/bsim/bluetooth/audio/test_scripts/cap_capture_and_render.sh | 1.681 | 20 |   | 0.08405
tests/bsim/bluetooth/audio/test_scripts/csip.sh | 2.881 | 30 |   | 0.0960333333333333
tests/bsim/bluetooth/audio/test_scripts/tbs.sh | 2.16 | 20 |   | 0.108
tests/bsim/bluetooth/audio/test_scripts/bap_broadcast_audio_assistant.sh | 5.221 | 50 |   | 0.10442
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_2.sh | 5.935 | 60 |   | 0.0989166666666667
tests/bsim/bluetooth/mesh/tests_scripts/beacon/beacon_interval.sh | 12.074 | 30 | 1 | 0.402466666666667
tests/bsim/bluetooth/audio/test_scripts/bap_bass_server_sync_server_rem.sh | 1.751 | 30 | 1 | 0.0583666666666667
tests/bsim/bluetooth/audio/test_scripts/csip_forced_release.sh | 3.286 | 30 |   | 0.109533333333333
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_mixed_fail.sh | 18.909 | 30 | 1 | 0.6303
tests/bsim/bluetooth/audio/test_scripts/pbp.sh | 4.43 | 50 |   | 0.0886
tests/bsim/bluetooth/audio/test_scripts/csip_no_size.sh | 4.731 | 30 |   | 0.1577
tests/bsim/bluetooth/audio/test_scripts/csip_no_rank.sh | 2.758 | 30 |   | 0.0919333333333333
tests/bsim/bluetooth/audio/test_scripts/mcs_mcc.sh | 4.063 | 50 |   | 0.08126
tests/bsim/bluetooth/audio/test_scripts/has_offline.sh | 1.203 | 20 |   | 0.06015
tests/bsim/bluetooth/audio/test_scripts/csip_new_sirk.sh | 2.394 | 30 |   | 0.0798
tests/bsim/bluetooth/audio/test_scripts/gmap_broadcast_ac_12.sh | 2.769 | 60 |   | 0.04615
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_4.sh | 5.924 | 60 |   | 0.0987333333333333
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_cli_persistent_transfer.sh | 24.513 | 30 | 1 | 0.8171
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_4.sh | 12.718 | 60 |   | 0.211966666666667
tests/bsim/bluetooth/audio/test_scripts/bap_unicast_audio.sh | 1.962 | 20 |   | 0.0981
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_persistent_transfer.sh | 26.593 | 30 | 1 | 0.886433333333333
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_5.sh | 14.084 | 60 |   | 0.234733333333333
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_11_i.sh | 15.736 | 60 |   | 0.262266666666667
tests/bsim/bluetooth/audio/test_scripts/bap_bass_client_sync.sh | 1.563 | 20 |   | 0.07815
tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_14.sh | 11.839 | 60 |   | 0.197316666666667
tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_13.sh | 13.318 | 60 |   | 0.221966666666667
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_10.sh | 13.111 | 60 |   | 0.218516666666667
tests/bsim/bluetooth/audio/test_scripts/csip_no_lock.sh | 2.814 | 30 |   | 0.0938
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_8_i.sh | 15.417 | 60 |   | 0.25695
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_trans_resume_push.sh | 31.032 | 30 | 1 | 1.0344
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_8_i.sh | 18.955 | 60 |   | 0.315916666666667
tests/bsim/bluetooth/audio/test_scripts/has.sh | 0.968 | 20 |   | 0.0484
tests/bsim/bluetooth/audio/test_scripts/gmap_broadcast_ac_13.sh | 3.177 | 60 |   | 0.05295
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_9_i.sh | 12.021 | 60 |   | 0.20035
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_2.sh | 12.25 | 60 |   | 0.204166666666667
tests/bsim/bluetooth/audio/test_scripts/media_controller.sh | 1.707 | 20 |   | 0.08535
tests/bsim/bluetooth/ll/edtt/tests_scripts/hci.llcp.sh | 1.111 | 30 | 1 | 0.0370333333333333
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_7_i.sh | 16.028 | 60 |   | 0.267133333333333
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_no_rsp_xfer.sh | 32.886 | 30 | 1 | 1.0962
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_trans_resume_pull.sh | 32.901 | 30 | 1 | 1.0967
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_5.sh | 14.495 | 60 |   | 0.241583333333333
tests/bsim/bluetooth/ll/advx/tests_scripts/basic_advx.sh | 2.073 | 120 |   | 0.017275
tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso_vs_dp.sh | 1.13 | 30 |   | 0.0376666666666667
tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn_encrypted_split_privacy.sh | 0.12 | 5 |   | 0.024
tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn20_split.sh | 0.294 | 60 |   | 0.0049
tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso.sh | 1.439 | 30 |   | 0.0479666666666667
tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn_encrypted_split_single_timer.sh | 0.127 | 5 |   | 0.0254
tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn_split_low_lat.sh | 0.074 | 5 |   | 0.0148
tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn_split.sh | 0.086 | 5 |   | 0.0172
tests/bsim/bluetooth/ll/conn/tests_scripts/basic_conn_encrypted_split.sh | 0.115 | 5 |   | 0.023
tests/bsim/bluetooth/ll/bis/tests_scripts/broadcast_iso_ticker_expire_info.sh | 1.545 | 30 |   | 0.0515
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_3.sh | 15.534 | 60 |   | 0.2589
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_6_ii.sh | 14.107 | 60 |   | 0.235116666666667
tests/bsim/bluetooth/ll/advx/tests_scripts/basic_advx_ticker_expire_info.sh | 2.014 | 120 |   | 0.0167833333333333
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_6_i.sh | 12.044 | 60 |   | 0.200733333333333
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_first_ft_per_skip_4_se.sh | 1.747 | 60 |   | 0.0291166666666667
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_first_ft_per_skip_2_se.sh | 1.806 | 60 |   | 0.0301
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_first_ft_cen_skip_2_se.sh | 1.899 | 60 |   | 0.03165
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_1.sh | 5.681 | 60 |   | 0.0946833333333333
tests/bsim/bluetooth/audio/test_scripts/bap_broadcast_audio.sh | 8.691 | 50 |   | 0.17382
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_3.sh | 17.236 | 60 |   | 0.287266666666667
tests/bsim/bluetooth/ll/throughput/tests_scripts/gatt_write.sh | 3.932 | 60 |   | 0.0655333333333333
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_6_i.sh | 6.137 | 60 |   | 0.102283333333333
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_first_ft_cen_skip_4_se.sh | 1.139 | 60 |   | 0.0189833333333333
tests/bsim/bluetooth/hci_uart/tests_scripts/basic_conn_split_hci_uart.sh | 0.807 | 20 |   | 0.04035
tests/bsim/bluetooth/audio/test_scripts/cap_broadcast_ac_12.sh | 12.065 | 60 |   | 0.201083333333333
tests/bsim/bluetooth/hci_uart/tests_scripts/basic_conn_split_hci_uart_async.sh | 0.984 | 20 |   | 0.0492
tests/bsim/bluetooth/ll/edtt/tests_scripts/gap.llcp.sh | 6.277 | 30 | 1 | 0.209233333333333
tests/bsim/bluetooth/host/privacy/legacy/test_scripts/run_test.sh | 0.896 | 30 | 1 | 0.0298666666666667
tests/bsim/bluetooth/host/l2cap/credits/tests_scripts/l2cap_credits_ecred.sh | 0.619 | 30 | 1 | 0.0206333333333333
tests/bsim/bluetooth/host/l2cap/credits/tests_scripts/l2cap_credits.sh | 0.611 | 30 | 1 | 0.0203666666666667
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_group_acl_first.sh | 4.794 | 200 |   | 0.02397
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_group.sh | 4.878 | 200 |   | 0.02439
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_1.sh | 13.346 | 60 |   | 0.222433333333333
tests/bsim/bluetooth/ll/edtt/tests_scripts/gatt.llcp.sh | 10.207 | 300 |   | 0.0340233333333333
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_6_ii.sh | 23.487 | 60 |   | 0.39145
tests/bsim/bluetooth/host/l2cap/frags/tests_scripts/run.sh | 0.143 | 30 | 1 | 0.00476666666666667
tests/bsim/bluetooth/host/l2cap/split/test_scripts/l2cap_split.sh | 0.3 | 30 | 1 | 0.01
tests/bsim/bluetooth/host/l2cap/userdata/tests_scripts/l2cap.sh | 0.3 | 120 |   | 0.0025
tests/bsim/bluetooth/host/l2cap/credits_seg_recv/tests_scripts/l2cap_credits_seg_recv_ecred.sh | 0.565 | 30 | 1 | 0.0188333333333333
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_peripheral_cis.sh | 5.823 | 100 |   | 0.05823
tests/bsim/bluetooth/mesh/tests_scripts/dfu/dfu_srv_persistence.sh | 41.931 | 30 | 1 | 1.3977
tests/bsim/bluetooth/host/l2cap/credits_seg_recv/tests_scripts/l2cap_credits_seg_recv.sh | 0.569 | 30 | 1 | 0.0189666666666667
tests/bsim/bluetooth/host/l2cap/send_on_connect/tests_scripts/l2cap.sh | 0.443 | 120 |   | 0.00369166666666667
tests/bsim/bluetooth/host/privacy/central/test_scripts/run_test.sh | 4.967 | 30 | 1 | 0.165566666666667
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_9_ii.sh | 24.324 | 60 |   | 0.4054
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_legacy_adv_acl_first.sh | 7.711 | 200 |   | 0.038555
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_acl_first.sh | 7.991 | 200 |   | 0.039955
tests/bsim/bluetooth/host/l2cap/general/tests_scripts/l2cap.sh | 3.87 | 120 |   | 0.03225
tests/bsim/bluetooth/host/privacy/peripheral/test_scripts/run_test_rpa_expired.sh | 5 | 30 | 1 | 0.166666666666667
tests/bsim/bluetooth/host/gatt/caching/test_scripts/gatt_caching_retry_reads_no_eatt.sh | 0.819 | 30 | 1 | 0.0273
tests/bsim/bluetooth/host/gatt/caching/test_scripts/gatt_caching_db_hash_read_eatt.sh | 0.824 | 30 | 1 | 0.0274666666666667
tests/bsim/bluetooth/host/gatt/caching/test_scripts/gatt_caching_db_hash_read_no_eatt.sh | 0.788 | 30 | 1 | 0.0262666666666667
tests/bsim/bluetooth/host/gatt/caching/test_scripts/gatt_caching_out_of_sync_eatt.sh | 0.821 | 30 | 1 | 0.0273666666666667
tests/bsim/bluetooth/host/gatt/caching/test_scripts/gatt_caching_out_of_sync_no_eatt.sh | 0.854 | 30 | 1 | 0.0284666666666667
tests/bsim/bluetooth/host/privacy/device/test_scripts/run_tests.sh | 4.992 | 30 | 1 | 0.1664
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_mixed_unenhanced.sh | 0.745 | 30 | 1 | 0.0248333333333333
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_unenhanced_none.sh | 0.741 | 30 | 1 | 0.0247
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_unenhanced_mixed.sh | 0.722 | 30 | 1 | 0.0240666666666667
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_unenhanced_enhanced.sh | 0.75 | 30 | 1 | 0.025
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_mixed_none.sh | 0.699 | 30 | 1 | 0.0233
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_none_none.sh | 0.74 | 30 | 1 | 0.0246666666666667
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_8_ii.sh | 28.694 | 60 |   | 0.478233333333333
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_enhanced_unenhanced.sh | 0.745 | 30 | 1 | 0.0248333333333333
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_11_ii.sh | 31.341 | 60 |   | 0.52235
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_none_unenhanced.sh | 0.733 | 30 | 1 | 0.0244333333333333
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_unenhanced_unenhanced.sh | 0.732 | 30 | 1 | 0.0244
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_none_mixed.sh | 0.653 | 30 | 1 | 0.0217666666666667
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_enhanced_enhanced.sh | 0.625 | 30 | 1 | 0.0208333333333333
tests/bsim/bluetooth/host/id/settings/test_scripts/settings.sh | 0.034 | 30 | 1 | 0.00113333333333333
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_none_enhanced.sh | 0.632 | 30 | 1 | 0.0210666666666667
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_mixed_mixed.sh | 0.638 | 30 | 1 | 0.0212666666666667
tests/bsim/bluetooth/host/misc/unregister_conn_cb/tests_scripts/unregister_conn_cb.sh | 0.111 | 30 | 1 | 0.0037
tests/bsim/bluetooth/host/misc/disable/tests_scripts/disable.sh | 0.058 | 120 |   | 0.000483333333333333
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso.sh | 9.579 | 200 |   | 0.047895
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_mixed_enhanced.sh | 0.67 | 30 | 1 | 0.0223333333333333
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_enhanced_none.sh | 0.668 | 30 | 1 | 0.0222666666666667
tests/bsim/bluetooth/host/att/sequential/test_scripts/sequential.sh | 0.168 | 30 | 1 | 0.0056
tests/bsim/bluetooth/host/gatt/notify/test_scripts/gatt_notify_enhanced_mixed.sh | 0.765 | 30 | 1 | 0.0255
tests/bsim/bluetooth/host/gatt/sc_indicate/test_scripts/sc_indicate.sh | 0.753 | 30 |   | 0.0251
tests/bsim/bluetooth/host/misc/disconnect/test_scripts/disconnect.sh | 0.476 | 30 | 1 | 0.0158666666666667
tests/bsim/bluetooth/host/gatt/general/test_scripts/gatt.sh | 0.938 | 120 |   | 0.00781666666666667
tests/bsim/bluetooth/host/gatt/authorization/test_scripts/gatt.sh | 0.857 | 120 |   | 0.00714166666666667
tests/bsim/bluetooth/host/gatt/notify_multiple/test_scripts/notify.sh | 0.945 | 120 |   | 0.007875
tests/bsim/bluetooth/host/att/read_fill_buf/test_scripts/run_tests.sh | 0.127 | 30 | 1 | 0.00423333333333333
tests/bsim/bluetooth/host/privacy/peripheral/test_scripts/run_test.sh | 7.468 | 30 | 1 | 0.248933333333333
tests/bsim/bluetooth/host/att/retry_on_sec_err/test_scripts/run_test_security_request.sh | 0.175 | 30 | 1 | 0.00583333333333333
tests/bsim/bluetooth/host/att/eatt/tests_scripts/reconfigure.sh | 0.56 | 120 |   | 0.00466666666666667
tests/bsim/bluetooth/host/att/retry_on_sec_err/test_scripts/run_test.sh | 0.183 | 30 | 1 | 0.0061
tests/bsim/bluetooth/host/privacy/peripheral/test_scripts/run_test_rpa_sharing.sh | 7.624 | 30 | 1 | 0.254133333333333
tests/bsim/bluetooth/host/adv/encrypted/ead_sample/test_scripts/ead_sample.sh | 0.186 | 30 | 1 | 0.0062
tests/bsim/bluetooth/host/att/mtu_update/test_scripts/run_test.sh | 0.829 | 30 | 1 | 0.0276333333333333
tests/bsim/bluetooth/host/adv/extended/tests_scripts/ext_adv_conn.sh | 0.19 | 30 | 1 | 0.00633333333333333
tests/bsim/bluetooth/host/adv/extended/tests_scripts/ext_adv.sh | 0.091 | 30 | 1 | 0.00303333333333333
tests/bsim/bluetooth/host/gatt/ccc_store/test_scripts/ccc_store.sh | 1.833 | 30 | 1 | 0.0611
tests/bsim/bluetooth/host/gatt/ccc_store/test_scripts/ccc_store_2.sh | 1.878 | 30 | 1 | 0.0626
tests/bsim/bluetooth/host/att/long_read/test_scripts/run.sh | 0.894 | 30 | 1 | 0.0298
tests/bsim/bluetooth/host/att/eatt/tests_scripts/collision.sh | 1.516 | 120 |   | 0.0126333333333333
tests/bsim/bluetooth/host/att/eatt/tests_scripts/lowres.sh | 1.49 | 20 |   | 0.0745
tests/bsim/bluetooth/host/adv/encrypted/css_sample_data/test_scripts/run_tests.sh | 0.528 | 30 | 1 | 0.0176
tests/bsim/bluetooth/host/adv/resume/test_scripts/run_adv_resume_2.sh | 0.309 | 30 | 1 | 0.0103
tests/bsim/bluetooth/host/att/eatt/tests_scripts/multiple_conn.sh | 1.583 | 120 |   | 0.0131916666666667
tests/bsim/bluetooth/host/adv/extended/tests_scripts/ext_adv_conn_x5.sh | 0.413 | 30 | 1 | 0.0137666666666667
tests/bsim/bluetooth/host/adv/chain/tests_scripts/adv_chain.sh | 0.164 | 30 | 1 | 0.00546666666666667
tests/bsim/bluetooth/host/att/eatt/tests_scripts/autoconnect.sh | 1.476 | 120 |   | 0.0123
tests/bsim/bluetooth/host/security/bond_overwrite_denied/test_scripts/run_test.sh | 0.36 | 30 | 1 | 0.012
tests/bsim/bluetooth/host/adv/resume/test_scripts/run_adv_resume.sh | 0.711 | 30 | 1 | 0.0237
tests/bsim/bluetooth/host/adv/periodic/tests_scripts/per_adv_long_data.sh | 0.896 | 30 | 1 | 0.0298666666666667
tests/bsim/bluetooth/host/adv/periodic/tests_scripts/per_adv.sh | 0.837 | 30 | 1 | 0.0279
tests/bsim/bluetooth/host/gatt/caching/test_scripts/gatt_caching_retry_reads_eatt.sh | 4.445 | 30 | 1 | 0.148166666666667
tests/bsim/bluetooth/host/adv/periodic/tests_scripts/per_adv_conn.sh | 0.982 | 30 | 1 | 0.0327333333333333
tests/bsim/bluetooth/host/security/id_addr_update/test_scripts/run_test.sh | 0.377 | 30 | 1 | 0.0125666666666667
tests/bsim/bluetooth/ll/cis/tests_scripts/connected_iso_legacy_adv.sh | 10.092 | 200 |   | 0.05046
tests/bsim/bluetooth/host/security/ccc_update/test_scripts/ccc_update_2.sh | 0.731 | 30 | 1 | 0.0243666666666667
tests/bsim/bluetooth/host/security/bond_per_connection/test_scripts/run_test.sh | 0.492 | 30 | 1 | 0.0164
tests/bsim/bluetooth/host/security/ccc_update/test_scripts/ccc_update.sh | 0.725 | 30 | 1 | 0.0241666666666667
tests/bsim/bluetooth/host/security/bond_overwrite_allowed/test_scripts/run_test.sh | 0.852 | 30 | 1 | 0.0284
tests/bsim/bluetooth/host/security/security_changed_callback/test_scripts/security_changed_callback.sh | 0.89 | 30 | 1 | 0.0296666666666667
tests/bsim/bluetooth/host/adv/periodic/tests_scripts/per_adv_conn_privacy.sh | 1.054 | 30 | 1 | 0.0351333333333333
tests/bsim/bluetooth/host/iso/cis/tests_scripts/cis.sh | 0.507 | 120 |   | 0.004225
tests/bsim/bluetooth/host/gatt/settings/test_scripts/run_gatt_settings.sh | 5.19 | 120 |   | 0.04325
tests/bsim/bluetooth/host/gatt/settings/test_scripts/run_gatt_settings_2.sh | 5.197 | 120 |   | 0.0433083333333333
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_7_ii.sh | 34.267 | 60 |   | 0.571116666666667
tests/bsim/bluetooth/audio_samples/unicast_audio_client/tests_scripts/unicast_client.sh | 0.816 | 100 |   | 0.00816
tests/bsim/bluetooth/host/att/eatt_notif/test_scripts/eatt_notif.sh | 2.704 | 120 |   | 0.0225333333333333
tests/bsim/bluetooth/host/misc/disable/tests_scripts/disable_with_gatt.sh | 3.468 | 120 |   | 0.0289
tests/bsim/bluetooth/host/att/pipeline/test_scripts/run.sh | 3.854 | 30 | 1 | 0.128466666666667
tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_802154.sh | 2.388 | 100 |   | 0.02388
tests/bsim/bluetooth/audio/test_scripts/cap_unicast_ac_7_ii.sh | 22.575 | 60 |   | 0.37625
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_11_i.sh | 18.297 | 60 |   | 0.30495
tests/bsim/bluetooth/host/att/open_close/test_scripts/run.sh | 5.028 | 60 |   | 0.0838
tests/bsim/net/sockets/echo_test/tests_scripts/echo_test_ot.sh | 3.601 | 100 |   | 0.03601
tests/bsim/bluetooth/host/misc/hfc/test_scripts/run.sh | 7.193 | 120 |   | 0.0599416666666667
tests/bsim/bluetooth/ll/edtt/tests_scripts/ll.1.llcp.sh | 22.463 | 30 | 1 | 0.748766666666667
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_11_ii.sh | 39.976 | 60 |   | 0.666266666666667
tests/bsim/bluetooth/audio_samples/broadcast_audio_sink/tests_scripts/broadcast_audio.sh | 9.438 | 200 |   | 0.04719
tests/bsim/bluetooth/audio/test_scripts/gmap_unicast_ac_8_ii.sh | 34.259 | 60 |   | 0.570983333333333
tests/bsim/bluetooth/host/l2cap/stress/tests_scripts/l2cap.sh | 18.349 | 240 |   | 0.0764541666666667
tests/bsim/bluetooth/host/l2cap/stress/tests_scripts/l2cap_syswq.sh | 16.796 | 240 |   | 0.0699833333333333
tests/bsim/bluetooth/ll/edtt/tests_scripts/ll.2.llcp.sh | 28.43 | 30 | 1 | 0.947666666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_trans_complete_pull.sh | 62.135 | 30 | 1 | 2.07116666666667
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_cli_trans_complete_push.sh | 65.565 | 30 | 1 | 2.1855
tests/bsim/bluetooth/mesh/tests_scripts/blob_mdls/blob_srv_persistence.sh | 81.888 | 30 | 1 | 2.7296
tests/bsim/bluetooth/ll/multiple_id/tests_scripts/multiple.sh | 99.385 | 1800 |   | 0.0552138888888889


</body>

</html>

aescolar commented:
And if you are curious this is the crappy hack to get that csv:
```diff
diff --git a/tests/bsim/run_parallel.sh b/tests/bsim/run_parallel.sh
index 80e74512ef3..716eba0dcaf 100755
--- a/tests/bsim/run_parallel.sh
+++ b/tests/bsim/run_parallel.sh
@@ -75,6 +75,7 @@ echo -n "" > $tmp_res_file
 if [ `command -v parallel` ]; then
   parallel '
   echo "<testcase name=\"{}\" time=\"0\">"
+  start=$(date +%s%N)
   {} $@ &> {#}.log
   if [ $? -ne 0 ]; then
     (>&2 echo -e "\e[91m{} FAILED\e[39m")
@@ -86,7 +87,11 @@ if [ `command -v parallel` ]; then
     echo "</testcase>"
     exit 1
   else
-    (>&2 echo -e "{} PASSED")
+    dur=$(($(date +%s%N) - $start))
+    dur_s=$(echo "scale=3; $dur/1000000000" | bc)
+    E_TIME=$(grep EXECUTE_TIMEOUT {} | cut -d = -f2)
+    [[ -z "$E_TIME" ]] && E_TIME="30;1"
+    (>&2 echo -e "{} ; $dur_s ; $E_TIME")
     rm {#}.log
     echo "</testcase>"
   fi

```

jori-nordic commented:
> in my machine

You have a beefy-ass machine my dude. Can you run that on the CI? would be interesting to see the difference.

alwa-nordic commented:
Can we commandeer @aescolar's machine for CI? :smile:

But seriously, [I've had to re-run the bsim job four times to get my PR green](https://github.com/zephyrproject-rtos/zephyr/actions/runs/8420040902?pr=70670). We are not exactly saving CI costs right now by having short timeouts. Please make the pain go away. This should have a hot-fix tag. We can tune the timeouts later.

aescolar commented:
> But seriously, [I've had to re-run the bsim job four times to get my PR green](https://github.com/zephyrproject-rtos/zephyr/actions/runs/8420040902?pr=70670). We are not exactly saving CI costs right now by having short timeouts. Please make the pain go away. This should have a hot-fix tag. We can tune the timeouts later.

Which tests failed? (or please give me a link to the PR). It is much easier that way.

Thalley commented:
> Please make the pain go away. This should have a hot-fix tag. We can tune the timeouts later.

I agree with this statement. 

We have an issue.
This PR fixes that issue.
It's being "blocked" as there *may* be *potential* other issues from increasing the timeout. 

I'd much rather just get the issue fixed, and then optimize the timeout later (if needed at all).

jori-nordic commented:
discussed offline: pref is to update each test when it times out on a PR
